core/src/main/java/org/bitcoinj/base/internal/ByteUtils.java

```diff
@@ -289,6 +289,13 @@ public static void writeUint32LE(long val, OutputStream stream) throws IOExcepti
         stream.write(buf);
     }
 
+    /**
+     * Write a 4-byte integer to the output stream in little endian format.
+     * The integer may be signed or unsigned, its bytes are just serialized as is.
```

Comment:
It's good that you mention both of these cases. However as a consumer of this method I have almost no idea what "its bytes are just serialized as is" means.

I'd like to know:

* Can I use it to serialize signed integers as `int`? – probably yes, as that's the bread-and-butter case for Java
* Can I use it to serialize unsigned `long` integers? – it really should, if you claim "the integer may be signed", because that's how Java supports unsigned int
* Can I use it to serialize unsigned `int`? – that case should be explained in more detail, it's not really expected

I think the JavaDoc should make the answers to the questions clear. It's my perspective as a consumer who doesn't really want to think about these things.

Reply:
"The integer may be signed or unsigned, in both cases it is serialized correctly."


> * Can I use it to serialize signed integers as int?

Yes.

> * Can I use it to serialize unsigned long integers?

No, because the parameter type is an `int`.


> * Can I use it to serialize unsigned int?

Yes, is the above text (" in both cases it is serialized correctly") sufficient.

Reply:
The wording "in both cases it is serialized correctly" is at least reassuring (-:

But I'm not happy with the "the integer may be signed or unsigned", as long as I cannot supply unsigned `long` integers. If you claim "the integer may be signed" I'd expect that. To fix that, I'd either add the appropriate overload or be very specifc about it only accepts the unsigned `int` form. I'd prefer the overload, I guess it could call though to the original method with a cast.

(I'm trying to make a case here, such that we can apply the outcome of this discussion to the other write helpers.)